packages/salesforcedx-vscode-replay-debugger/src/states/variableAssignmentState.ts

```diff
@@ -164,21 +162,20 @@ export class VariableAssignmentState implements DebugLogState {
       Object.keys(obj).forEach(key => {
         const refContainer = logContext.getRefsMap().get(String(obj[key]))!;
         if (refContainer) {
-          const tmpContainer = new ApexVariableContainer(
+          const tmpContainer = this.copyReferenceContainer(
+            refContainer,
             key,
-            refContainer.value,
-            refContainer.type,
-            refContainer.ref
+            logContext
           );
-          tmpContainer.variables = refContainer.variables;
-          tmpContainer.variablesRef = logContext
-            .getVariableHandler()
-            .create(tmpContainer);
           container.variables.set(key, tmpContainer);
         } else {
+          let varValue = obj[key];
+          if (typeof varValue === 'string') {
```

Comment:
If the value is in a json object, check if the parsed value is a string and then append single quotes